variety/PreferencesVarietyDialog.py

```diff
@@ -748,8 +749,12 @@ def edit_source(self, edited_row):
             if type == Options.SourceType.FLICKR:
                 self.dialog = AddFlickrDialog()
             elif type in Options.CONFIGURABLE_IMAGE_SOURCES_MAP:
-                self.dialog = AddConfigurableDialog()
-                self.dialog.set_source(Options.CONFIGURABLE_IMAGE_SOURCES_MAP[type])
+                if type == Options.SourceType.WALLHAVEN:
```

Comment:
this can be moved up and made symmetrical to the case for FLICKR, or the FLICKR case can be moved within the body of `elif type in Options.CONFIGURABLE_IMAGE_SOURCES_MAP`. Not sure, but also `self.dialog.set_source(Options.CONFIGURABLE_IMAGE_SOURCES_MAP[type])` could probably be called only once, regardless of the type.

Reply:
>or the FLICKR case can be moved within the body of elif type in Options.CONFIGURABLE_IMAGE_SOURCES_MAP.

It seems like that Options.CONFIGURABLE_IMAGE_SOURCES_MAP doesn't contains Options.SourceType.FLICKR.

>but also self.dialog.set_source(Options.CONFIGURABLE_IMAGE_SOURCES_MAP[type]) could probably be called only once, regardless of the type.

Indeed, done.